src/file/format/yaml.rs

```diff
@@ -20,8 +21,26 @@ pub fn parse(
             return Err(Box::new(MultipleDocumentsError(n)));
         }
     };
-
-    // TODO: Have a proper error fire if the root of a file is ever not a Table
+    match root {
+        yaml::Yaml::String(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Str(value)))
+        }
+        yaml::Yaml::Real(value) => {
+            return Err(ConfigError::invalid_root(
+                uri,
+                Unexpected::Float(value.parse::<f64>().unwrap_or(0.0)),
+            ))
+        }
+        yaml::Yaml::Integer(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Integer(value)))
+        }
+        yaml::Yaml::Boolean(value) => {
+            return Err(ConfigError::invalid_root(uri, Unexpected::Bool(value)))
+        }
+        yaml::Yaml::Hash(_) => (),
+        yaml::Yaml::Array(_) => return Err(ConfigError::invalid_root(uri, Unexpected::Seq)),
+        _ => return Err(ConfigError::invalid_root(uri, Unexpected::Unit)),
+    };
```

Comment:
I'm not sure why you're changing yaml stuff in a PR for json5, please remove this change.